src/tree/registries/azure/AzureAccountTreeItem.ts

```diff
@@ -28,4 +30,33 @@ export class AzureAccountTreeItem extends AzureAccountTreeItemBase implements IR
     public createSubscriptionTreeItem(subContext: ISubscriptionContext): SubscriptionTreeItem {
         return new SubscriptionTreeItem(this, subContext);
     }
+
+    public async loadMoreChildrenImpl(_clearCache: boolean, context: IActionContext): Promise<AzExtTreeItem[]> {
+        const treeItems: AzExtTreeItem[] = await super.loadMoreChildrenImpl(_clearCache, context);
+
+        if (treeItems.length === 1 && treeItems[0].commandId === 'extension.open') {
+            // tslint:disable-next-line: no-floating-promises
+            this.refreshNodeWhenAzureExtensionIsInstalled();
+        }
+
+        return treeItems;
+    }
+
+    private async refreshNodeWhenAzureExtensionIsInstalled(): Promise<void> {
+        let isExtensionInstalled: boolean = false;
+        let timedout: boolean = false;
+        const maxTimeout: number = 600000; //10 minutes
+        setTimeout(() => { timedout = true; }, maxTimeout);
```

Comment:
Ideally this timeout timer will get canceled when an installation happens.

Reply:
Yes, if we can actually have a handler for the installation of this particular extension, then we do not need to poll at all. Can we?

Reply:
Hmm, the [`vscode.extensions.onDidChange`](https://code.visualstudio.com/api/references/vscode-api#extensions) event looks promising.  We'd still need to ensure we're only listening at most once (see above comment), but would be better than polling.

Reply:
Now using event onDidChange. But still each instance need to subscribe to the event.